src/OS8104A.ts

```diff
@@ -0,0 +1,697 @@
+import { Gpio } from "onoff"
+import spi, { type SpiDevice, type SpiOptions } from "spi-device"
+import EventEmitter from "events"
+import { getRegisterConfig } from "./RegisterConfig"
+import { Registers } from "./Registers.js"
+const TRANSFER_SPEED = 180000
+
+export interface MostMessage {
+    targetAddressHigh: number
+    targetAddressLow: number
+    fBlockID: number
+    instanceID: number
+    fktId: number
+    opType: number
+    data: number[]
+}
+
+export interface Config {
+    freq: number
+    nodeAddressLow: number
+    nodeAddressHigh: number
+    groupAddress: number
+}
+
+export interface Stream {
+    sourceAddrHigh: number
+    sourceAddrLow: number
+    fBlockID: number
+    instanceID: number
+    sinkNr: number
+}
+
+export interface AllocResult {
+    loc1: number
+    loc2: number
+    loc3: number
+    loc4: number
+    cl: number
+    answer1: string
+    freeChannels: number
+}
+
+export interface SourceResult {
+    nodePos: number
+    group: number
+    logicalHigh: number
+    logicalLow: number
+}
```

Comment:
originally had these as types, however eslint was adamant that an interface is correct here, need to research some more.

Reply:
Config change resolved this, moved all types to Messages.ts

https://github.com/rhysmorgan134/SocketMost/blob/typescript/src/modules/Messages.ts